### ISSUE TITLE
[Flex Video] Semantic Name & Definable Ratio 

### ISSUE THREAD

**Issue by andycochran** [zurb/foundation-sites#8199]
I'd like to propose a couple changes to Flex Video because: 
- The `.flex-video` class can actually be used to responsively embed any type of `iframe`, `object`, `embed`, or `video`. Could be a map, an image gallery, a custom widget… 
- "Flex" might lead to confusion with flexbox in the documentation. And documentation clarity seems to be a concern. 

I propose we rename it "Responsive Embed."

We could allow either the `.flex-video` or `.responsive-embed` class, recommend the new name, and deprecate the old name in a future version. 

Also it would be nice if the ratio could be handled with a `data-` attribute to set the appropriate padding inline. For example… 

We could do:

``` html
<div class="responsive-embed" data-embed-ratio="16:9">
```

instead of: 

``` html
<div class="flex-video widescreen">
```

Would a PR for this be desirable?

**Comment by akshaybharwani** [comment]
I agree with this. `.responsive-embed` does sound good for the purpose.

**Comment by colin-marshall** [comment]
Should we introduce this in 6.3 with `.flex-video` deprecated, and completely remove `.flex-video` in 6.4?

**Comment by brettsmason** [comment]
I agree a rename to responsive-embed is a good idea and would make more sense :+1: . Just need to make sure its deprecated in a way that makes sense and nice and clear in the docs.

In terms of adding data attributes I'm not sure how that would work, but I'm open to see how that could work. I currently have [this](https://github.com/brettsmason/croft/blob/master/assets/js/theme/theme.js#L12) is Croft for adding the correct wrapper to embeds if thats any use.

**Comment by coreysyms** [comment]
Just to chime in: this makes a lot of sense to me... I actually just had a developer of mine reference "flex-box" thinking "foundation's flex" was the same thing. Using this new class, whatever it ends up being called to wrap and responsively iframe's is a ticket I can get behind.

**Comment by andycochran** [comment]
@brettsmason, yeah, the data attribute part was a little half-baked. But we could add JavaScript to do what I did in @JeremyEnglert's https://github.com/JeremyEnglert/JointsWP/pull/191. 

Maybe something as simple as: 

```js
$('.flex-video iframe').each(function() {
  if ( $(this).innerWidth() / $(this).innerHeight() > 1.5 ) {
    $(this).parent('.flex-video').addClass('widescreen');
  }
});
```

But it'd be nice to have more options than the two ratio settings—`$flexvideo-ratio` (4:3) and `$flexvideo-ratio-widescreen ` (16:9). JavaScript could inject `style="padding-bottom: [ratio]%;"` inline on the `.flex-video` element. Then the width and height attributes on the `iframe` could explicitly determine the `.flex-video` element's padding. 

@Owlbertz, how 'bout the following? (You could probably clean this up.)

```js
$('.flex-video iframe').each(function() {
  var flex = $(this).parent('.flex-video');
  flex.removeClass('flex-video');
  var ratio = $(this).innerHeight() / $(this).innerWidth() * 100 + '%'; 
  flex.addClass('flex-video').css('padding-bottom', ratio );
});
```

**Comment by Owlbertz** [comment]
Yeah, renaming would be appropriate.
Regarding the code: this seems to not work well for me somehow... Adding this to the [docs page](http://foundation.zurb.com/sites/docs/flex-video.html) causes even the widescreen video to become 75% padded... Do I miss something? Haven't really used this feature until now.

**Comment by andycochran** [comment]
@Owlbertz, you have to also change the `<iframe width="400" height="225" … ` to `<iframe width="560" height="315" … `. Currently, the second video on that page is 16:9, but the width/height values are set to 4:3. Should work on the third video tho.

**Comment by Owlbertz** [comment]
I see... Looks good for me. Where do you want to put that code?

**Comment by andycochran** [comment]
@Owlbertz I'll add it to a 6.3 PR. Just wanted to run the JS by you first.

**Comment by kball** [comment]
I like this direction a lot, thought I'd love it if we could keep it entirely scss.  What about introducing an scss map or array of sizes similar to how colors have been moved into a map?

**Comment by andycochran** [comment]
@kball, the JS could be optional and the CSS could continue to work for the default (4:3) and `.widescreen` (16:9).

**Comment by andycochran** [comment]
Closing this. It'll be in 6.3. 🎉

**Comment by andycochran** [comment]
Ack! Actually, only the semantics are addressed in 6.3. Reopening 'cause I'd like to get the dynamic sizes in there.
